src/mjml/mjml.service.js

```diff
@@ -43,14 +43,19 @@ export default class MjmlService {
 
   /**
    * Transform MJML to HTML
-   * @todo show validation erros in the UI
-   * @returns string
+   * @todo show validation errors in the UI
+   * @returns ?string
    */
-  static mjmlToHtml(mjml) {
+  static mjmlToHtml(mjml, endpoint = '') {
+    if (typeof mjml !== 'string' || !mjml.includes('<mjml>')) {
+      throw new Error('No valid MJML provided');
```

Comment:
did you move it out of the catch block? I think it will not work anymore. now whenever there is a syntax error it will break the JS code and the UI will freeze.

Reply:
I moved into the catch block again